src/renderer/components/commandPalette/index.vue

```diff
@@ -0,0 +1,385 @@
+<template>
+  <div class="command-palette">
+    <el-dialog
+      :visible.sync="showCommandPalette"
+      :show-close="false"
+      :modal="true"
+      @close="handleDialogClose"
+      custom-class="ag-dialog-table"
+      width="500px"
+    >
+      <div slot="title" class="search-wrapper">
+        <div class="input-wrapper">
+          <input
+            ref="search"
+            type="text"
+            v-model="query"
+            class="search"
+            @keydown="handleBeforeInput"
+            @keyup="handleInput"
+            placeholder="Type a command to execute"
+          >
+        </div>
+        <loading v-if="searcherBusy"></loading>
+        <transition name="fade" v-else-if="availableCommands.length">
+          <ul class="commands">
+            <li
+              v-for="(item, index) of availableCommands"
+              :key="index"
+              ref="command-items"
+              @click="search(item.id)"
+              :class="{'active': index === selectedCommandIndex}"
+            >
+              <span class="title" :title="item.title">{{item.description}}</span>
+              <span class="shortcut">
+                <span
+                  class="shortcut"
+                  v-for="(accelerator, index) of item.shortcut"
+                  :key="index"
+                >
+                    <kbd>{{accelerator}}</kbd>
+                </span>
+              </span>
+            </li>
+          </ul>
+        </transition>
+      </div>
+    </el-dialog>
+  </div>
+</template>
+
+<script>
+import { mapState } from 'vuex'
+import log from 'electron-log'
+import bus from '../../bus'
+import loading from '../aidou/loading'
+
+export default {
+  components: {
+    loading
+  },
+  computed: {
+    ...mapState({
+      rootCommand: state => state.commandCenter.rootCommand
+    })
+  },
+  data () {
+    this.currentCommand = null
+    return {
+      showCommandPalette: false,
+      query: '',
+      selectedCommandIndex: -1,
+      availableCommands: [],
+      searcherBusy: false
+    }
+  },
+  created () {
+    this.$nextTick(() => {
+      bus.$on('show-command-palette', this.handleShow)
+    })
+  },
+  beforeDestroy () {
+    bus.$off('show-command-palette', this.handleShow)
+  },
+  methods: {
+    handleShow (command) {
```

Comment:
@fxha I have a suggestion, I prefer set the input's placeholder to `go to file` or `search file to open` when the command is `file.quick-open`. or we can show different command with different input' placeholder.

```html
<input :placeholder="placeholder">
```

Reply:
We can show a short description of current command in the placeholder?